components/engine_rocks/src/table_properties.rs

```diff
@@ -9,6 +9,12 @@ impl engine_traits::UserCollectedProperties for UserCollectedProperties {
     fn get(&self, index: &[u8]) -> Option<&[u8]> {
         self.0.get(index)
     }
+
+    fn approximate_size_and_keys(&self, start: &[u8], end: &[u8]) -> Option<(usize, usize)> {
+        let rp = RangeProperties::decode(&self.0).ok()?;
```

Comment:
Where will this new method be called? This seems expensive.

Reply:
Here: https://github.com/tikv/tikv/pull/11385/files#r755931248.